test/tests/smoke_test.rs

```diff
@@ -362,3 +362,82 @@ fn smoke_test() {
 
     // TODO: Validate the rest of the fmc_alias certificate fields
 }
+
+#[test]
+fn fips_self_test() {
+    let fuses = Fuses::default();
+    let rom = caliptra_builder::build_firmware_rom(&ROM_WITH_UART).unwrap();
+    let mut hw = caliptra_hw_model::new(BootParams {
+        init_params: InitParams {
+            rom: &rom,
+            security_state: SecurityState::from(fuses.life_cycle as u32),
+            ..Default::default()
+        },
+        fuses,
+        ..Default::default()
+    })
+    .unwrap();
+
+    let image_bundle = caliptra_builder::build_and_sign_image(
+        &FMC_WITH_UART,
+        &APP_WITH_UART,
+        ImageOptions::default(),
+    )
+    .unwrap();
+
+    hw.tracing_hint(false);
+
+    hw.upload_firmware(&image_bundle.to_bytes().unwrap())
+        .unwrap();
+
+    hw.step_until_boot_status(
+        caliptra_runtime::RtBootStatus::RtReadyForCommands.into(),
+        true,
+    );
+
+    let payload = MailboxReqHeader {
+        chksum: caliptra_common::checksum::calc_checksum(u32::from(CommandId::SELF_TEST), &[]),
+    };
+
+    let resp = hw
+        .mailbox_execute(u32::from(CommandId::SELF_TEST), payload.as_bytes())
+        .unwrap()
+        .unwrap();
+
+    let resp = MailboxRespHeader::read_from(resp.as_slice()).unwrap();
+    // Verify checksum and FIPS status
+    assert!(caliptra_common::checksum::verify_checksum(
+        resp.chksum,
+        0x0,
+        &resp.as_bytes()[core::mem::size_of_val(&resp.chksum)..],
+    ));
+    assert_eq!(resp.fips_status, MailboxRespHeader::FIPS_STATUS_APPROVED);
+
+    // Confirm we can't re-start the FIPS self test.
+    hw.step_until_boot_status(
+        caliptra_runtime::RtBootStatus::RtFipSelfTestStarted.into(),
+        true,
+    );
+    let _resp = hw
+        .mailbox_execute(u32::from(CommandId::SELF_TEST), payload.as_bytes())
+        .unwrap_err();
+
+    hw.step_until_boot_status(
```

Comment:
I don't think it is realistic to expect callers to do this. Also, I'm not sure the boot status register is really intended to be used once we've fully booted.

Perhaps the mailbox response from the SELF_TEST command should explain to the user that the test is currently in progress.

Maybe it would be simpler if this were two separate commands `CommandId::SELF_TEST_START` and `CommandID:SELF_TEST_GET_RESULT`?

Reply:
done